workbench_utils.py

```diff
@@ -2772,8 +2773,30 @@ def download_remote_file(config, url):
         print('Error: ' + message)
 
     # create_media() references the path of the downloaded file.
-    downloaded_file_path = os.path.join(config['input_dir'], url.split("/")[-1])
-    open(downloaded_file_path, 'wb+').write(response.content)
+    subdir = config['input_dir'] + '/' + re.sub('[^A-Za-z0-9]+', '_', node_csv_row[config['id_field']])
+    Path(subdir).mkdir(parents=True, exist_ok=True)
+
+    if config["use_node_title_for_media"]:
```

Comment:
If 'use_node_title_for_media' is true, media based on files that have an extension aren't created. workbench log contains entries like `ERROR - Media not created, PUT request to "http://97.107.189.30/node/81/media/file/17" returned an HTTP status code of "400".`

Reply:
@ajstanley found this problem ^^ when the `file` column contains a combination of URLs to Islandora datastreams (your new code works for thees) and files that have an extension (media not created). I'm happy to merge in your code and troubleshoot this myself if you want. Let me know your preference.